lib/datadog/appsec/waf.rb

```diff
@@ -541,29 +519,24 @@ def required_addresses
 
         def merge(data)
           data_obj = Datadog::AppSec::WAF.ruby_to_object(data, coerce: false)
-          ruleset_info = LibDDWAF::RuleSetInfo.new
-          new_handle = Datadog::AppSec::WAF::LibDDWAF.ddwaf_update(handle_obj, data_obj, ruleset_info)
+          ruleset_info_obj = LibDDWAF::Object.new
+          new_handle = Datadog::AppSec::WAF::LibDDWAF.ddwaf_update(handle_obj, data_obj, ruleset_info_obj)
 
           return if new_handle.null?
 
-          info = {
-            loaded: ruleset_info[:loaded],
-            failed: ruleset_info[:failed],
-            errors: WAF.object_to_ruby(ruleset_info[:errors]),
-            version: ruleset_info[:version],
-          }
-          new_from_handle(new_handle, info, config)
+          ruleset_info = Datadog::AppSec::WAF.object_to_ruby(ruleset_info_obj)
+          new_from_handle(new_handle, ruleset_info, config)
         ensure
           Datadog::AppSec::WAF::LibDDWAF.ddwaf_object_free(data_obj) if data_obj
-          Datadog::AppSec::WAF::LibDDWAF.ddwaf_ruleset_info_free(ruleset_info) if ruleset_info
+          Datadog::AppSec::WAF::LibDDWAF.ddwaf_object_free(ruleset_info_obj) if ruleset_info_obj
         end
 
         private
 
-        def new_from_handle(handle_object, info, config)
+        def new_from_handle(handle_object, ruleset_info, config)
           obj = self.class.allocate
           obj.instance_variable_set(:@handle_obj, handle_object)
-          obj.instance_variable_set(:@ruleset_info, info)
+          obj.instance_variable_set(:@ruleset_info, ruleset_info)
           obj.instance_variable_set(:@config, config)
           obj
```

Comment:
Minor: This method is kinda weird, since it looks like an instance method and it's actually not.

Reply:
Here is the PR that introduced to provide some context https://github.com/DataDog/libddwaf-rb/pull/23